### DISCUSSION TITLE
NPM install from Dockerfile into the workspaces directory

### DISCUSSION THREAD

**Discussion by davisg123** [community/community#15284] [Codespaces]
I'd like to run `npm install` from my Dockerfile and not as a post install action in order to utilize prebuilt images

Is the `workspaces` directory used defined as an env variable anywhere? Or is the recommended way just to hardcode it?

```
WORKDIR /workspaces/{REPO_NAME}

RUN npm install -g yarn --quiet

RUN yarn install

```

**Comment by bartr** [comment]
Pre-build might work. put your npm and yarn command in on-create and turn on pre-build. We clone other repos to /workspaces in on-create and it works - not sure if it will work in the repo you're cloning or not.
